src/main/java/core/basesyntax/services/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.handlers.BalanceHandler;
+import core.basesyntax.strategy.handlers.OperationHandler;
+import core.basesyntax.strategy.handlers.PurchaseHandler;
+import core.basesyntax.strategy.handlers.ReceiveHandler;
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.OperationType, OperationHandler> map = Stream.of(
+            Map.entry(FruitTransaction.OperationType.BALANCE, new BalanceHandler()),
+            Map.entry(FruitTransaction.OperationType.RETURN, new ReceiveHandler()),
+            Map.entry(FruitTransaction.OperationType.PURCHASE, new PurchaseHandler()),
+            Map.entry(FruitTransaction.OperationType.SUPPLY, new ReceiveHandler())
+            ).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
create this map in the main and putt it as a parameter, also use just `Map.of()` it's improve your code readability